src/node.cc

```diff
@@ -1159,6 +1153,32 @@ Maybe<bool> ProcessEmitDeprecationWarning(Environment* env,
                                    deprecation_code);
 }
 
+static void OnMessage(Local<Message> message, Local<Value> error) {
+  Isolate* isolate = Isolate::GetCurrent();
+  switch (message->ErrorLevel()) {
+    case Isolate::MessageErrorLevel::kMessageWarning: {
+      Environment* env = Environment::GetCurrent(isolate);
+      if (!env) {
+        break;
+      }
+      v8::String::Utf8Value filename(isolate,
+          message->GetScriptOrigin().ResourceName());
+      // (filename):(line) (message)
```

Comment:
From a design/readability perspective, since `ProcessEmitWarningGeneric` just sends this back to JS, I'd do something like:
```cpp
Local<Value> args[3] = {
  filename,
  message->GetLineNumber(env->context()).FromMaybe(-1),
  message->Get()
};
if (env.format_warning.As<Function>()->Call(env->context(),
                                            process,
                                            arraysize(args),
                                            args).IsEmpty()) {
  oops;
}
```
and do the formatting in JS...
That way we can turn this comment into a template
```js
process.format_warning = function (filename, line, message) {
  if (check_some_option) {
    console.warn(`${filename}:${line} ${message});
  }
}
```

Reply:
I don't see any compelling reason to special case this, it's just a warning.

Reply:
The special case is it's structured, and we're clumsily flattening it in C++.
What I don't love is that here the formatting is forced (unwrapping the `Value`s and hiding the declarative nature of the template in imperative code).

But yeah it's not a blocker.

Reply:
Wait we have `ProcessEmitWarning(Environment* env, const char* fmt, ...)` so how about
```cpp
USE(ProcessEmitWarning(env, "%s:%d %s",
                       filename, 
                       message->GetLineNumber(env->context()).FromMaybe(-1),
                       msg));
```

Reply:
Oooofff, it has no `type` param.
NM.